# Simulation mode
simulation_mode: false             # Is set in launch file!
gazebo_simulation: true
### Kinematic configuration parameter

# Debug mode
activate_output: true
activate_debug_output: true
activate_controller_node_output: true
plotting_result: true

# Clock frequency //hz
clock_frequency: 25

#Optimal control problem dimensions
state_dim: 5
control_dim: 2

# Input and output topics
publish:
    cmd:          '/lmpcc'
    cmd_sim:      '/prius'
robot_state_topic: '/base_pose_ground_truth'
obs_state_topic: '/future_obj_track'
waypoint_topic: '/path_publisher/path' 

# Kinematic chain parameter
robot_base_link: base_link
target_frame: map # world fixed frame but will be changed according to robot_state_topic
global_path_frame: local_car

# Weight on the slack variable for collision collision
slack_weight: 10000
repulsive_weight: 0.1
reference_velocity: 2.0
initial_speed: 0.0

# Parameters for predefined reference path
global_path:
  x: [0,         50, 100,      230, 200, 100]
  y: [0,         0, -25,      -100, 0, 0]
  theta: [0, 0,  1.5708, 1.5708, 0, 0]
  n_points_clothoid: 20
  n_points_spline: 20

#Road parameters
road_width_right: 2.15
road_width_left: 5.125

# Parameters for collision constraints
obstacles:
      n_obstacles: 2
      sub_ellipse_topic: ellipse_objects_feed
      n_discs: 3
      ego_l: 4.540
      ego_w: 1.760

constraints:
     velocity_constraints:
           min: [0]
           max: [13.8]
     #orientation_constraints:

acado_config:
     max_num_iteration: 70
     kkt_tolerance: 1e-3
     integrator_tolerance: 1e-3
     start_time_horizon: 0.0
     end_time_horizon: 2.0 # 5.0  # control and prediction horizon
     discretization_intervals: 50
     sampling_time: 0.04
     weight_factors:
           contour_weight_factors:
                #Wcontour and Wlag
                [0.1, 0.1]
           control_weight_factors:
                # Wa and Wdelta
                [1.0, 4.0]
                
